### PR TITLE
[superseded] niminst: make linenoise.h optional

### PR DESCRIPTION
**Pull request by timotheecour**
refs: https://github.com/nim-lang/Nim/commit/2b368bcdd705a5c7f0558df706fde01d217f5e9d#commitcomment-37273026
/cc @drjdn does this work for you?

### CODE REVIEW

File: tools/niminst/niminst.nim [hunk 1]
```diff
@@ -513,7 +513,10 @@ template gatherFiles(fun, libpath, outDir) =
       fun(src, dst)
 
     for f in walkFiles(libpath / "lib/*.h"): copySrc(f)
-    copySrc(libpath / "lib/wrappers/linenoise/linenoise.h")
```
Comment (Araq) [inline tools/niminst/niminst.nim:516]: Seriously? This is your fix? "let's always copy linenoise.h except if it fails we don't mind"?

File: tools/niminst/niminst.nim [hunk 2]
```diff
@@ -144,23 +146,30 @@ const
...
-  -m, --main:file     set the main nim file, by default ini-file with .nim
-                      extension
-  --var:name=value    set the value of a variable
-  -h, --help          shows this help
-  -v, --version       shows the version
+  -o, --output:dir       set the output directory
+  -m, --main:file        set the main nim file, by default ini-file with .nim
+                         extension
+  --var:name=value       set the value of a variable
+  --withLinenoise:on|off whether to include linenoise.h, on by default
```
Comment (Araq) [inline tools/niminst/niminst.nim]: 1. It should be ``off`` by default.
2. The switch should be more general like ``--bundleNim`` for Nim compiler specific logic.

**Comment by timotheecour** [comment]
adding a flag to niminst (eg `--bundleNim`) and then passing it from `./koch csources` ended up causing a rabbit hole of complexity/extra code just for that 1 boolean flag. Instead I've changed the code so that this works:
```
./koch csource -d:release -d:nimBundleNimExtra
```
and made a reference to that in ./koch --help

**Comment by drjdn** [comment]
@timotheecour I can confirm that this now works again as expected.

**Comment by Araq** [comment]
With this PR nightlies will be broken once again. Can we please revert this linenoise patch and make ``nim secret`` its own tool. It's also blocking my merge of https://github.com/nim-lang/Nim/pull/13065 !!

**Comment by drjdn** [comment]
@timotheecour could you please back out your modifications to niminst.  Your last changes fix my initial issue but I'm onside with @Araq that this should be a separate tool.  Currently we have the following in: https://github.com/nim-lang/Nim/blob/3dad130034e3212159dda5d61988ca2b18959b24/compiler/options.nim#L695
which basically means if you want to use nim's compiler internals and export that to standalone C code you need to copy around a header for a line editing library that is very unlikely to be needed.  Exporting C code like this is one of nim's killer features in my opinion.  Until an actual REPL that handles the whole language and FFI exists (which would be awesome) I don't see why this needs to be in the compiler directory and definitely not in `niminst`.

**Comment by timotheecour** [comment]
all right, see https://github.com/nim-lang/Nim/pull/13478 ; if it gets merged we can close this

**Comment by narimiran** [comment]
> if it gets merged we can close this

Since it has `merge_when_passes_CI` label, I'm closing this one already.

**Pull request closed without merging**